test/error-messages/constraint-arg.txt

```diff
@@ -0,0 +1,9 @@
+Constraint violation
+
+semiFn :: Semigroup a => a -> a
+          ^^^^^^^^^^^    ^
+                         1
+
+1) 42 :: Number
+
+‘semiFn’ requires ‘a’ to implement Semigroup.
```

Comment:
What do you think of this tweak?

```diff
-‘semiFn’ requires ‘a’ to implement Semigroup.
+‘semiFn’ requires ‘a’ to satisfy the Semigroup type-class constraint.
+The value at position 1 does not satisfy the constraint, therefore
+‘a’ does not satisfy the constraint.
```

On a related note, do you think we should hard-wrap long lines? It would make messages easier to read in a wide terminal. Were we to wrap at 79 characters, say, we'd risk unfortunate line wrapping in a slightly narrower terminal.

I'm slightly in favour of hard wrapping.